kubecost/templates/cluster-controller/cluster-controller-deployment.yaml

```diff
@@ -174,8 +182,17 @@ spec:
           # and their validity.
           optional: true
       {{- if not (.Values.clusterController).legacyMode }}
+      {{- if and (not (.Values.clusterController).storageConfigSecret) (not (.Values.clusterController).storageConfig) }}
+      - name: config-storage
+        emptyDir: { }
+      - name: federated-storage-config
+        secret:
+          defaultMode: 420
+          secretName: {{ include "kubecost.federatedStorage.secretName" . }}
```

Comment:
We assume that this secret will exist; we assume that the cluster controller is installed alongside aggregator and so the `kubecost.federatedStorage.secretName` (ie. `.values.global.federatedStorage.existingSecret` or the config to create the secret) will already have been defined.

Reply:
yes, safe assumption, and ok to explode if this is not the case